src/authz-smart/smartHandler.ts

```diff
@@ -8,31 +8,130 @@ import {
     AuthorizationRequest,
     AuthorizationBundleRequest,
     AllowedResourceTypesForOperationRequest,
+    SystemOperation,
+    TypeOperation,
 } from 'fhir-works-on-aws-interface';
-import { SMARTConfig } from './smartConfig';
+import axios from 'axios';
+import { LaunchType, ScopeType, SMARTConfig } from './smartConfig';
 
 // eslint-disable-next-line import/prefer-default-export
 export class SMARTHandler implements Authorization {
+    static readonly SCOPE_REGEX = /(patient|user|system|launch)\/(\w+|\*)\.?(\w+|\*)?|(^launch$)/;

```

Comment:
What is the system scope?

Why is`'read' | 'write'` not explicitly on the regex? `patient/Observation.doSomething` shouldn't match

Use anchors `^$`, `SomePreffixpatient/Patient.readSomeSuffix` should not match 

It may be better to use separate regex for clinical scopes and launch scopes or separate them in a more explict way (like `<clinical regex>|<launch regex>`). Right now this matches invalid weird combinations like `launch/*.read`

It may be worth having unit tests specifically for the `isScopeSufficient` function

Reply:
System scope -> http://build.fhir.org/ig/HL7/bulk-data-export/authorization/index.html#scopes

+1 on regex fixes

Adding unit tests for isScopeSufficient means it will need to be public (not the end of the world but a bit odd)